### ISSUE TITLE
Abundance_difference does not work when unpooled

### ISSUE THREAD

**Issue by SallyKoerner** [NCEAS/codyn#95]
Ok if I run this
AbunDiff<-abundance_difference(df = ramps3,
                            species.var = "species",
                            abundance.var = "cov",
                            replicate.var = "uniqueID",
                            treatment.var = "P_H",
                            time.var = "year",
                            pool = TRUE)
it works.

However, when I run
AbunDiff<-abundance_difference(df = ramps3,
                            species.var = "species",
                            abundance.var = "cov",
                            replicate.var = "uniqueID",
                            treatment.var = "P_H",
                            time.var = "year",
                            pool = FALSE,
                            block.var=NULL)
I get this error:
Error in `[.data.frame`(df, as.character(args[grep("\\.var$", names(args))])) : 
  undefined columns selected

OR if I run this
AbunDiff<-abundance_difference(df = ramps3,
                            species.var = "species",
                            abundance.var = "cov",
                            replicate.var = "uniqueID",
                            treatment.var = "P_H",
                            time.var = "year")
I get this error:
There were 11 warnings (use warnings() to see them)
> warnings()
Warning messages:
1: In Ops.factor(cross[[cross.var]], cross[[cross.var2]]) :
  ‘<’ not meaningful for factors
2: In Ops.factor(cross[[cross.var]], cross[[cross.var2]]) :
  ‘<’ not meaningful for factors
3: In Ops.factor(cross[[cross.var]], cross[[cross.var2]]) :
  ‘<’ not meaningful for factors
4: In Ops.factor(cross[[cross.var]], cross[[cross.var2]]) :
  ‘<’ not meaningful for factors
5: In Ops.factor(cross[[cross.var]], cross[[cross.var2]]) :
  ‘<’ not meaningful for factors
6: In Ops.factor(cross[[cross.var]], cross[[cross.var2]]) :
  ‘<’ not meaningful for factors
7: In Ops.factor(cross[[cross.var]], cross[[cross.var2]]) :
  ‘<’ not meaningful for factors
8: In Ops.factor(cross[[cross.var]], cross[[cross.var2]]) :
  ‘<’ not meaningful for factors
9: In Ops.factor(cross[[cross.var]], cross[[cross.var2]]) :
  ‘<’ not meaningful for factors
10: In Ops.factor(cross[[cross.var]], cross[[cross.var2]]) :
  ‘<’ not meaningful for factors
11: In Ops.factor(cross[[cross.var]], cross[[cross.var2]]) :
  ‘<’ not meaningful for factors

basically I cannot do it UNPOOLED and when I type code in that should work either way, I get two different errors.

Thanks

**Comment by SallyKoerner** [comment]
I get the same issue with Rac_difference and curve_difference. Same exact errors when I run the unpooled code in the two different ways

**Comment by mavolio** [comment]
@itcarroll here is the spreadsheet that Sally used to get this error.
[ramps.txt](https://github.com/NCEAS/codyn/files/1991703/ramps.txt)

**Comment by itcarroll** [comment]
Hi @SallyKoerner,

Thanks for catching one bug! I fixed the first error message. But, I cannot reproduce the warning messages (`< not meaningful for factors`). Can you say exactly how you read the ramps.txt file into R? I am wondering if UniqueID is a special data type.

Thanks,
Ian

**Comment by itcarroll** [comment]
Closing: unable to reproduce.
